src/charm.py

```diff
@@ -199,6 +226,61 @@ def _restart(self, event: EventBase):
 
         self.container.restart(CHARM_KEY)
 
+        # Indicate that unit has completed restart
+        self.cluster.relation.data[self.unit]["password-rotated"] = "true"
+
+        # If leader runs last on RollingOps restart, this code would be enough
+        """
+        if self.unit.is_leader() and self.cluster.relation.data[self.app].get("rotate-passwords"):
+            logger.error("LEADER REMOVING ROTATE_PASSWORDS")
+            self.cluster.relation.data[self.app]["rotate-passwords"] = ""
+        """
+
+    def _get_super_password_action(self, event: ActionEvent) -> None:
+        """Handler for get-super-password action event."""
+        event.set_results({"super-password": self.cluster.passwords[0]})
+
+    def _get_sync_password_action(self, event: ActionEvent) -> None:
+        """Handler for get-sync-password action event."""
+        event.set_results({"sync-password": self.cluster.passwords[1]})
+
+    def _set_password_action(self, event: ActionEvent) -> None:
+        """Handler for set-password action.
+
+        Set the password for a specific user, if no passwords are passed, generate them.
+        """
+        if not self.unit.is_leader():
+            msg = "Password rotation must be called on leader unit"
+            logger.error(msg)
+            event.fail(msg)
+            return
+
+        username = "super"
+        if "username" in event.params:
+            username = event.params["username"]
+        if username not in CHARM_USERS:
```

Comment:
Am I understanding that this only rotates passwords for super/sync users, and that we aren't permitting changing of passwords for, say, `relation-5` user?

Reply:
yes, only operator users at the moment.